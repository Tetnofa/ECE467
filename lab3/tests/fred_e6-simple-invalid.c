int main() {
  // This also tests that the line number of the error 
  // reported is the line number corresponding to 
  // the LEFT operand (for some reason);
  (true + false) || true; // Invalid: '+' not defined for bools
}
