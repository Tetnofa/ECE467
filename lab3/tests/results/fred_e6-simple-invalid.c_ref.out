Program [useIO=0](1,0)
	Function[isProto=0]	Int(1,0)
		Identifier[name:main](1,4)
		Scope(1,11)
			Expression Statement(5,2)
				Bool Expression(5,2)
					Binary Expression[opcode:||](5,2)
						Int Expression(5,3)
							Binary Expression[opcode:+](5,3)
								Int Expression(5,3)
									BoolConstant[val=1](5,3)
								Int Expression(5,10)
									BoolConstant[val=0](5,10)
						Int Expression(5,20)
							BoolConstant[val=1](5,20)
sema: 5:3 : type mismatch 
sema: 5:3 : type mismatch 
